.github/workflows/ci-pipeline.yml

```diff
@@ -16,4 +16,20 @@ jobs:
     steps:
     - uses: actions/checkout@v4
     - run: docker build --target test --tag todo-app:test .
-    - run: docker run todo-app:test
\ No newline at end of file
+    - run: docker run todo-app:test
+
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    needs: build
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main'}}
+    steps:
+    - uses: actions/checkout@v4
+    - name: Login to Docker Hub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKER_HUB_USERNAME }}
+        password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+    - run: docker build --target production --tag chiaradinansoftwire/my-todo-app:latest .
+    - run: docker push chiaradinansoftwire/my-todo-app:latest
+    - run: curl -dH -X POST '${{ secrets.APP_SERVICE_DEPLOYMENT_WEBHOOK }}'
```

Comment:
I'm a big fan of adding a `--fail` argument here, because without it curl defaults to assuming that any legitimate response is a success, where _legitimate_ here implies that curl has done it's job in getting a message to the address and receiving a response, but that response might be a 500 "Everything has blown up" response! `--fail` tells it to at least consider this command to have failed if a 4xx or 5xx status code comes back